compiler/src/fuzzer/values.rs

```diff
@@ -0,0 +1,206 @@
+use std::mem;
+
+use crate::{
+    builtin_functions,
+    vm::{Data, Heap, List, Packet, Pointer, Symbol},
+};
+use itertools::Itertools;
+use num_bigint::RandBigInt;
+use rand::{
+    prelude::ThreadRng,
+    seq::{IteratorRandom, SliceRandom},
+    Rng,
+};
+use rustc_hash::FxHashMap;
+
+use super::utils::Input;
+
+pub fn generate_input(num_args: usize, symbols: &[String]) -> Input {
+    let mut arguments = vec![];
+    for _ in 0..num_args {
+        arguments.push(generate_value(symbols));
+    }
+    Input { arguments }
+}
+
+fn generate_value(symbols: &[String]) -> Packet {
+    let mut heap = Heap::default();
+    let address =
+        generate_value_with_complexity(&mut heap, &mut rand::thread_rng(), 100.0, symbols);
+    Packet { heap, address }
```

Comment:
We could join the heaps of multiple arguments, so that `Input` only has a `Heap` and a `Vec<Pointer>`